bugfix alpine glacier

luraess opened the pull request:
The `fetch_glacier` function was broken wrt alpine glacier selection. This PR fixes following:
- capitalise `OST` and `WEST` to match downloaded swissTLM3D data file names
- use `joinpath` for more robust path handling
- use `DimPoints` to retrieve DEM coordinates
- use updated Rasters.jl to fix `mosaic` function 
- HDF5 I/O

luraess committed:
repair

luraess committed:
fix raster to array handling and more

luraess committed:
update CI

mauro3 commented on src/Alpine_glaciers.jl:
```diff
@@ -17,27 +17,28 @@ function fetch_glacier(name::String, SGI_ID::String; datadir::String)
+                     joinpath.(dir,"08_SurfaceElevation_SwissAlps/",readdir(joinpath(dir,"08_SurfaceElevation_SwissAlps/")))
+                     joinpath.(dir,"TLM_BB", filter(x->startswith(x,"swissTLM3D_TLM_BODEN") && (endswith(x,".shp") || endswith(x,".dbf")),readdir(joinpath(dir,"TLM_BB/"))))
+                     joinpath.(dir,"TLM_BB", filter(x->startswith(x,"swissTLM3D_TLM_GLAMOS") && endswith(x,".dbf"),readdir(joinpath(dir,"TLM_BB/"))))]
+    dirs_to_remove = joinpath.(dir,filter(x->isdir(dir*x),readdir(dir)))
```
use `joinpath` here as well

luraess commented:
Actually still not OK as it now hangs on `mask(rasterDat; with=poly)` ...

mauro3 reviewed (commented):
Some tests would be good too.  Especially as Rasters is a moving target.

mauro3 committed:
Fixed fetch_glacier.

Note that mask_trim was very inefficient.  Now cropping is faster with
`crop_padded`.

mauro3 committed:
joinpath

mauro3 commented:
Have a look at the code.  I think that does what you want it to do.

luraess commented:
> Have a look at the code. I think that does what you want it to do.

Indeed - that does what I want. Thanks

mauro3 commented:
Note that the padding is now in meters and not gridpoints. (oops: edit "not" -> "now")

luraess commented:
Will try to add some tests.

mauro3 commented:
Merge this now?

luraess commented:
I'd say so, and I'll open a new issue to add tests.
